modulemd/v2/modulemd-buildopts.c

```diff
@@ -45,6 +45,36 @@ enum
 static GParamSpec *properties[N_PROPS];
 
 
+gboolean
+modulemd_buildopts_equals (ModulemdBuildopts *self_1,
+                           ModulemdBuildopts *self_2)
+{
+  if (!self_1 && !self_2)
+    return TRUE;
+
+  if (!self_1 || !self_2)
+    return FALSE;
+
+  g_return_val_if_fail (MODULEMD_IS_BUILDOPTS (self_1), FALSE);
+  g_return_val_if_fail (MODULEMD_IS_BUILDOPTS (self_2), FALSE);
+
+  if (!modulemd_hash_table_sets_are_equal (self_1->whitelist,
+                                           self_2->whitelist))
+    return FALSE;
+
+  if (!self_1->rpm_macros && !self_2->rpm_macros)
```

Comment:
You don't need to do these NULL-checks because `g_strcmp0()` is NULL-aware. It will do this for you.